src/crowbar.mli

```diff
@@ -1,52 +1,118 @@
-type state
-
-type 'a printer = Format.formatter -> 'a -> unit
+(** {1:top Types } *)
 
 type 'a gen
+(** an ['a gen] knows how to generate ['a] for use in Crowbar tests. *)
 
 type ('k, 'res) gens =
   | [] : ('res, 'res) gens
   | (::) : 'a gen * ('k, 'res) gens -> ('a -> 'k, 'res) gens
+(** multiple generators are provided to functions using a listlike syntax.
+    for example, [map [int; int] (fun a b -> a + b)] *)
+
+type 'a printer = Format.formatter -> 'a -> unit
+(** pretty-printers for items generated by Crowbar; useful for the user in
+    translating test failures into bugfixes. *)
+
+(**/**)
 (* re-export stdlib's list
    We only want to override [] syntax in the argument to Map *)
 type nonrec +'a list = 'a list = [] | (::) of 'a * 'a list
+(**/**)
 
-val map : ('f, 'a) gens -> 'f -> 'a gen
+(** {1:top Generators } *)
+
+(** {2 Simple Generators } *)
 
+val int : int gen
+(** [int] generates an integer.  For 32-bit systems, it will generate an
+    Int32.t; otherwise, it will generate an Int64.t .  If you need integers from
+    a smaller domain, consider using {!range}. *)
+val uint8 : int gen
```

Comment:
I think these docstring attachement are ambiguous (add one more newline at the end of comments) see https://github.com/xavierleroy/cryptokit/issues/3#issuecomment-328047667 for reference.

Reply:
Thanks!  The [manual](https://github.com/ocaml/ocaml/blob/trunk/manual/manual/cmds/ocamldoc.etex#L504) claims this is OK, but these do trigger warning 50 (and it's easier to read them in `vim` with the newlines, too).